conjure-runtime/src/body.rs

```diff
@@ -175,7 +175,10 @@ where
 {
     fn poll_fill_buf(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<io::Result<&[u8]>> {
         while !self.cur.has_remaining() {
-            match ready!(self.as_mut().project().body.poll_data(cx)).transpose()? {
+            match ready!(self.as_mut().project().body.poll_data(cx))
+                .transpose()
+                .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?
+            {
```

Comment:
The old gzip implementation already used [`io::Error`](https://doc.rust-lang.org/stable/std/io/struct.Error.html), while the new one uses the more general `Box<dyn Error>`. Since the Read trait we're implementing here requires io::Error, we have to convert at this point.

Reply:
This diff specifically: https://github.com/palantir/conjure-rust-runtime/pull/142/files#diff-e2226fd1e4d3d9d9bcf944d682b9ed0581d26158e7a4597748bb7ca27a155a90L130